utilities/resource_sync/sync.py

```diff
@@ -618,19 +650,72 @@ def main():
+                for job_name in job_names:
+                    res = src_glue.get_job(JobName=job_name)
+                    jobs.append(res['Job'])
+            else:
+                get_jobs_paginator = src_glue.get_paginator('get_jobs')
+                for page in get_jobs_paginator.paginate():
+                    jobs.extend(page['Jobs'])
+            
+            save_jobs_to_file(jobs, args.serialize_file)
+            logger.info(f"Jobs serialized to file: {args.serialize_file}")
+            return  # Exit after serialization
+
+        elif args.deploy_file:
+            # Deploy jobs from file
+            if not args.config_path:
+                logger.error("--config-path must be provided when using --deploy-from-file")
+                sys.exit(1)
+            
+            jobs = load_jobs_from_file(args.deploy_file)
+            logger.info("Applying mapping transformations to jobs")
+            jobs = apply_mapping_to_jobs(jobs, mapping)
+            
+            for job in jobs:
+                job_name = job['Name']
+                try:
+                    logger.debug(f"Checking if job '{job_name}' exists in the destination account.")
+                    current_job = dst_glue.get_job(JobName=job_name)
+                    if args.overwrite_jobs:
+                        del job['Name']
+                        job_update = {'JobName': job_name, 'JobUpdate': job}
+                        logger.debug(f"Updating job '{job_name}' with configuration: '{json.dumps(job_update, indent=4, default=str)}'")
+                        if do_update:
+                            dst_glue.update_job(**job_update)
+                        logger.info(f"The job '{job_name}' has been overwritten.")
+                except dst_glue.exceptions.EntityNotFoundException:
+                    logger.debug(f"Creating job '{job_name}' with configuration: '{json.dumps(job, indent=4, default=str)}'")
+                    if do_update:
+                        dst_glue.create_job(**job)
+                    logger.info(f"New job '{job_name}' has been created.")
+                except Exception as e:
+                    logger.error(f"Error occurred in deploying job: '{job_name}'")
+                    if args.skip_errors:
+                        logger.error(f"Skipping error: {e}", exc_info=True)
+                    else:
+                        raise
+
+        else:
+            # Existing synchronization logic
+            if args.src_job_names is not None:
+                logger.debug(f"Sync target job: {args.src_job_names}")
+                job_names = args.src_job_names.split(',')
+                for job_name in job_names:
+                    synchronize_job(job_name, mapping)
+            else:
+                jobs = []
+                get_jobs_paginator = src_glue.get_paginator('get_jobs')
+                for page in get_jobs_paginator.paginate():
+                    jobs.extend(page['Jobs'])
```

Comment:
The same logic is already used in L656-664. Can we clean up the code to avoid defining the same logic twice?